### DISCUSSION TITLE
coord.npy

### DISCUSSION THREAD

**Discussion by ayda1** [CSIprinceton/workshop-july-2022#6] [General]
Into a frame, what exactly does the coordinates file (coord.npy ) represent and how do I get them?

**Comment by PabloPiaggi** [comment]
Hi @ayda1 , you can find a good description of the format of ```coord.npy``` [here](https://docs.deepmodeling.com/projects/deepmd/en/master/data/data-conv.html). The file contains the atomic coordinates of a system (number of frames * 3 * number of atoms). You can prepare this file in raw (plain text) format in any way you want to. I use [ase](https://wiki.fysik.dtu.dk/ase/) to extract data from the output of electronic structure calculations, but you can also use [dpdata](https://github.com/deepmodeling/dpdata). Once you have the data in raw format (```coord.raw```), you can convert it to numpy using the ```raw_to_set.sh``` script found in ```$deepmd_source_dir/data/raw/raw_to_set.sh```

**Reply by ayda1** [reply]
Thank you for your reply
Additionally please , what exactly does the box.raw represent and how do I get them?

**Reply by PabloPiaggi** [reply]
Did you check the explanation in the manual? box.raw contains the box vectors. You can get it in the same way I described for coord.raw . Let me know if you have other, more specific, doubts

**Reply by ayda1** [reply]
yes I checked the manuel and Box is a vector of dimension (Nframes * 3 * 3) in the order XX XY XZ YX
YY YZ ZX ZY ZZ.  But I want to know if the values of the vectors are generated randomly. If not how it is obtain.

The frame below come to my input.data. how to get from the frame the coordinate of the vector?

begin
atom           0.000000000         0.000000000         0.000000000 O         0.000000000         0.000000000         0.000000000         0.000000000         0.000000000
atom           1.000000000         0.000000000         0.000000000 H         0.000000000         0.000000000         0.000000000         0.000000000         0.000000000
atom           0.283662185        -0.958924275         0.000000000 H         0.000000000         0.000000000         0.000000000         0.000000000         0.000000000
energy         2.09399700
charge         0.00000000
end

For water example in deep-md the box contains the following repeatly
 [12.444661,  0.      ,  0.      ,  0.      , 12.444661,  0. ,  0.      ,  0.      , 12.444661],
How to get this value?

I am very confuse
Thanks for consideration

**Reply by PabloPiaggi** [reply]
Hi @ayda1 , I need more context to understand your question. How are you generating the input data for the model that you are trying to train? Are you using DFT? If so, which code? QE, Vasp, CP2K?

**Reply by ayda1** [reply]
I did not get my input.data neither from CP2K nor quantum espresso
I have a surface in the form of a fortran-code.
I generated this surface in cartesian coordinates and I obtained the attached data file.
my job is from my input.data to generate the files suitable for deep-md using the NumPy format
[(https://docs.deepmodeling.com/projects/deepmd/en/master/data/data-conv.html]
I generated the coord.raw, energy.raw files according to my input.dat and I could not generate the box.raw file.

**Reply by PabloPiaggi** [reply]
Is your system periodic? Do you have the lengths of each side of the simulation box?

**Reply by ayda1** [reply]
my system is water 
what do you mean - lengths on each side of the simulation box

**Reply by PabloPiaggi** [reply]
If you are not familiar with the idea of a simulation box, please read a book on molecular simulation such as "Understanding Molecular Simulation" by Frenkel and Smit, or ask someone. These are basic concepts and it doesn't make sense to discuss them here.

**Reply by ayda1** [reply]
Thanks, I will check it out
